src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -765,7 +769,39 @@ class InvokeHostFunctionApplyHelper : virtual LedgerAccessHelper
         mOpFrame.innerResult(mRes).code(INVOKE_HOST_FUNCTION_SUCCESS);
         mOpFrame.innerResult(mRes).success() = xdrSha256(success);
 
-        mOpMeta.getEventManager().setEvents(std::move(success.events));
+        if (!mProtocol23SACReconciliationEvents.empty())
+        {
+            xdr::xvector<ContractEvent> events;
+            events.reserve(success.events.size() +
+                           mProtocol23SACReconciliationEvents.size());
+            for (auto const& rEvent : mProtocol23SACReconciliationEvents)
+            {
+                if (rEvent.amount > 0)
+                {
+                    events.emplace_back(mOpMeta.getEventManager().makeMintEvent(
+                        rEvent.asset, rEvent.mintOrBurnAddress, rEvent.amount,
+                        false));
+                }
+                else
+                {
+                    events.emplace_back(mOpMeta.getEventManager().makeBurnEvent(
+                        rEvent.asset, rEvent.mintOrBurnAddress,
+                        -rEvent.amount));
+                }
+                CLOG_WARNING(Ledger,
```

Comment:
All of the log messages in this PR should be reduced to INFO, and the messages themselves need to be cleaned up.